babel.config.js

```diff
@@ -0,0 +1,85 @@
+module.exports = function (api) {
+  var validEnv = ["development", "test", "production"];
+  var currentEnv = api.env();
+  var isDevelopmentEnv = api.env("development");
+  var isProductionEnv = api.env("production");
+  var isTestEnv = api.env("test");
+
+  if (!validEnv.includes(currentEnv)) {
+    throw new Error(
+      "Please specify a valid `NODE_ENV` or " +
+        '`BABEL_ENV` environment variables. Valid values are "development", ' +
+        '"test", and "production". Instead, received: ' +
+        JSON.stringify(currentEnv) +
+        "."
+    );
+  }
+
+  return {
+    presets: [
+      isTestEnv && [
+        "@babel/preset-env",
+        {
+          targets: {
+            node: "current",
+          },
+          modules: "commonjs",
+        },
+        "@babel/preset-react",
+      ],
+      (isProductionEnv || isDevelopmentEnv) && [
+        "@babel/preset-env",
+        {
+          forceAllTransforms: true,
+          useBuiltIns: "entry",
+          corejs: 3,
+          modules: false,
+          exclude: ["transform-typeof-symbol"],
+        },
+      ],
+      [
+        "@babel/preset-react",
+        {
+          development: isDevelopmentEnv || isTestEnv,
+          useBuiltIns: true,
+        },
+      ],
+    ].filter(Boolean),
+    plugins: [
+      ["@babel/plugin-proposal-private-methods"],
+      ["@babel/plugin-proposal-private-property-in-object"],
+      ["@babel/plugin-proposal-class-properties"],
+      "babel-plugin-macros",
+      "@babel/plugin-syntax-dynamic-import",
+      isTestEnv && "babel-plugin-dynamic-import-node",
+      "@babel/plugin-transform-destructuring",
+      [
+        "@babel/plugin-proposal-object-rest-spread",
+        {
+          useBuiltIns: true,
+        },
+      ],
+      [
+        "@babel/plugin-transform-runtime",
+        {
+          helpers: false,
+          regenerator: true,
+          corejs: false,
+        },
+      ],
+      [
+        "@babel/plugin-transform-regenerator",
+        {
+          async: false,
+        },
+      ],
+      isProductionEnv && [
+        "babel-plugin-transform-react-remove-prop-types",
+        {
+          removeImport: true,
+        },
+      ],
+      isDevelopmentEnv && "react-refresh/babel",
+    ].filter(Boolean),
+  };
+};
```

Comment:
It is better to restrict the babel config to what is required. If the `.babelrc` file is encountered during rebase (while resolving conflicts), please delete `.babelrc` and keep `babel.config.js` file.

```suggestion
module.exports = api => {
  api.cache(true);

  return {
    presets: ["@babel/preset-env", "@babel/preset-react"],
    plugins: ["@babel/plugin-proposal-class-properties"],
    env: {
      production: {
        plugins: [
          "@babel/plugin-proposal-class-properties",
          "@babel/plugin-transform-runtime",
        ],
      },
    },
  };
};

```